### PR TITLE
Exploration/26 UI prototype pm integration

### PR DESCRIPTION
**Pull request by nils-schmitt**
To have a common baseline, we want to pull this branch onto main. All mergeconflicts are resolved and pylint is happy.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Content directory
-content/inputs/
-content/outputs/intermediates/
-content/outputs/new_prompts.txt
+tracex/extraction/content/inputs
```
Comment (PitButtchereit) [inline .gitignore:2]: Isn't the content folder seperate to the traxec django project folder?

File: .pylintrc [hunk 2]
```diff
@@ -11,6 +11,8 @@ disable=too-many-arguments,
   fixme,
   import-error,
   unspecified-encoding,
+  too-few-public-methods,
+  import-outside-toplevel,
 
 [FORMAT]
 max-line-length=100
```
Comment (PitButtchereit) [inline .pylintrc:18]: I would suggest increasing the max-line-lenght to 200 for fewer restrictions.

File: tracex/extraction/content/outputs/all_traces_event_type_1.xes
Comment (PitButtchereit) [inline tracex/extraction/content/outputs/all_traces_event_type_1.xes:1]: why are there two files like this?

File: tracex/extraction/content/outputs/intermediates/1_bulletpoints.txt
Comment (PitButtchereit) [inline tracex/extraction/content/outputs/intermediates/1_bulletpoints.txt:1]: shouldn't this file be ignored from git?

File: tracex/extraction/forms.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        help_text=f"Allowed file types: {', '.join(ALLOWED_FILE_TYPES)}",
+        validators=[FileExtensionValidator(allowed_extensions=ALLOWED_FILE_TYPES)],
+        error_messages={
+            "required": "Please provide a file from which to extract the event log."
+        },
+    )
+    field_order = ["journey", "event_types", "locations"]
+
+
+class GenerationForm(BaseEventForm):
```
Comment (PitButtchereit) [inline tracex/extraction/forms.py:73]: This class will be implemented later?

File: tracex/extraction/prototype/create_prompts.py
Comment (PitButtchereit) [inline tracex/extraction/prototype/create_prompts.py:1]: is this file still needed?

File: tracex/extraction/prototype/utils.py [hunk 4]
```diff
@@ -2,20 +2,25 @@
...
+from django.conf import settings
 
 import openai
 
-out_path = Path("content/outputs/")  # Path to the outputs-folder
-in_path = Path("content/inputs/")  # Path to the inputs-folder
+out_path = settings.BASE_DIR / Path(
+    "extraction/content/outputs/"
+)  # Path to the outputs-folder
+in_path = settings.BASE_DIR / Path(
```
Comment (PitButtchereit) [inline tracex/extraction/prototype/utils.py:12]: variable names should be no abbreviations

**Review by PitButtchereit — CHANGES_REQUESTED** [review]
Still some unclear things for me. Please explain on comments

File: .gitignore (hunk 1 above)
Comment (FR-SON) [reply .gitignore:2]: Django really doesn't like accessing files from outside its own scope. So we moved it inside the scope.

File: .pylintrc (hunk 2 above)
Comment (FR-SON) [reply .pylintrc:18]: 200 seems too long to me. The standard for PyCharm is 120, is that a suitable compromise?

File: tracex/extraction/content/outputs/all_traces_event_type_1.xes
Comment (FR-SON) [reply tracex/extraction/content/outputs/all_traces_event_type_1.xes:1]: The naming is not clear, but this file contains the single trace, that was extracted. The other one is the actual "all_traces" event log. Name is given by the function we currently use. This will be changed during the next refactoring, so a filename can be set as a parameter when calling the function.

File: tracex/extraction/content/outputs/intermediates/1_bulletpoints.txt
Comment (FR-SON) [reply tracex/extraction/content/outputs/intermediates/1_bulletpoints.txt:1]: yes, it should

File: tracex/extraction/forms.py (hunk 3 above)
Comment (FR-SON) [reply tracex/extraction/forms.py:73]: This class inherits from BaseEventForm and adds no further functionality, its the same. Currently that's only for readability so we know what form we are working with. That should be discussed and possibly changed.

File: tracex/extraction/prototype/create_prompts.py
Comment (FR-SON) [reply tracex/extraction/prototype/create_prompts.py:1]: Why wouldn't this file be needed? But I hand this question to @nils-schmitt

File: tracex/extraction/prototype/create_all_trace_xes.py [hunk 5]
```diff
@@ -32,15 +32,16 @@ def create_all_trace_xes(csv_file, key):
...
         activity_key="concept:name",
         timestamp_key="time:timestamp",
     )
+    return str(u.out_path / output_name)
 
 
-ACTIVITY_KEY = get_key()
-create_all_trace_xes(u.CSV_OUTPUT, ACTIVITY_KEY)
+# ACTIVITY_KEY = 'event_type' # get_key()
+# create_all_trace_xes(u.CSV_OUTPUT, ACTIVITY_KEY)
```
Comment (FR-SON) [inline tracex/extraction/prototype/create_all_trace_xes.py:47]: This needs to be resolved. Theses lines where a problem in the UI, but seem to necessary in the command line tool. We need to find a way to make the execution of this function not mandatory.
Maybe the call can be made explicitly in main?

**Review by FR-SON — CHANGES_REQUESTED** [review]
Everything seems fine, aside from the one additional "change" I requested. The rest either looks good or is already scheduled to be refactored later on.

File: .pylintrc (hunk 2 above)
Comment (nils-schmitt) [reply .pylintrc:18]: I also think, that 120 should be enough.

File: tracex/extraction/prototype/create_prompts.py
Comment (nils-schmitt) [reply tracex/extraction/prototype/create_prompts.py:1]: It is more likely a testing file, but i could imagine using it in the pipeline and even if not, possibly as an addition or beforehand the existing process

File: tracex/extraction/prototype/create_all_trace_xes.py (hunk 5 above)
Comment (nils-schmitt) [reply tracex/extraction/prototype/create_all_trace_xes.py]: As we do not work with xes in the terminal variant of the pipeline, we never call get_key anyway. I would remove two lines, but keep the function as it could come in handy when creating different dfgs later on

File: tracex/extraction/content/outputs/intermediates/1_bulletpoints.txt
Comment (nils-schmitt) [reply tracex/extraction/content/outputs/intermediates/1_bulletpoints.txt:1]: Unfortunately this works only for newly created files in that directories...

File: tracex/extraction/content/outputs/all_traces_event_type_1.xes
Comment (PitButtchereit) [reply tracex/extraction/content/outputs/all_traces_event_type_1.xes:1]: I created a ticket for this

File: tracex/extraction/content/outputs/intermediates/1_bulletpoints.txt
Comment (PitButtchereit) [reply tracex/extraction/content/outputs/intermediates/1_bulletpoints.txt:1]: @FR-SON yes

File: tracex/extraction/forms.py (hunk 3 above)
Comment (PitButtchereit) [reply tracex/extraction/forms.py:73]: ticket created

**Review by PitButtchereit — APPROVED** [review]
all good from my side

File: tracex/extraction/prototype/create_all_trace_xes.py (hunk 5 above)
Comment (FR-SON) [reply tracex/extraction/prototype/create_all_trace_xes.py]: Very well, we might also use this, once we provide the option to select the activity key in the UI.

**Review by FR-SON — APPROVED** [review]

File: tracex/extraction/content/outputs/all_traces_event_type_1.xes
Comment (nils-schmitt) [reply tracex/extraction/content/outputs/all_traces_event_type_1.xes:1]: No need, it's done. You now call "create_xes()" with an activity_key and name. The result gets saved as 
name + "_" + activity_key + ".xes"
